### ISSUE TITLE
[Feature Request] Winston enable receiving objects in the logs

### ISSUE THREAD

**Issue by lveillard** [highlight/highlight#7096]
Hello, with the configuration proposed in the docs for winston:

```
const highlightTransport = () => {
	return new transports.Http({
		host: 'pub.highlight.run',
		path: '/v1/logs/json',
		ssl: true,
		headers: {
			'x-highlight-project': '...',
			'x-highlight-service': '...',
		},
		level: 'success',
		format: format.combine(format.json(), format.errors({ stack: true }), format.timestamp(), format.prettyPrint()),
	});
};
```

When doing things like this
```ts
logger.info("User authed", myUser)
```

The objects in the second position don't get sent to the logger, so we are forced to do:

```ts
logger.info(`User auther: ${JSON.stringify(myUser)}`)
```

It would be amazing if the objects added as "payload" into the logger could arrive also like that in the logger so we could inspect them in the highlight frontent without getting ugly long texts.

Also, not sure if is related but while it is active in preprod we get some nextjs logs and others we dont, for intance this:
![image](https://github.com/highlight/highlight/assets/29461076/1d5b7ff1-cfae-4dc9-b424-2b07c874448e)

Arrives in hughlight's frontend as this

![image](https://github.com/highlight/highlight/assets/29461076/0454d93c-99f2-4aa9-a978-d6050ecc512e)

I wonder if the second line has also a different format that makes highligh ignore it?

**Comment by jay-khatri** [comment]
Hi, this is Highlight.io's [GitSorted](https://gitsorted.onrender.com/) bot. 
                While one of our developers cooks up a reply, please search for anything related in our [Discord](https://highlight.io/community) or on our [docs](https://highlight.io/docs). 
                Also, if you haven't posted a reproduction, please do so (we prioritize those tickets).

**Comment by Vadman97** [comment]
![Screenshot from 2023-12-12 14-07-11](https://github.com/highlight/highlight/assets/1351531/d31774ee-43e5-4b6d-bf32-05c57c049edc)

![Screenshot from 2023-12-12 14-05-12](https://github.com/highlight/highlight/assets/1351531/d76ed7e7-2b13-46b9-90f5-d3d6cb694c59)

@lveillard not sure i can reproduce this. i'm getting all attributes sent over correctly. here's my setup
```tsx
const highlightTransport = new winston.transports.Http({
	host: 'pub.highlight.run',
	path: '/v1/logs/json',
	ssl: true,
	headers: {
		'x-highlight-project': projectID,
		'x-highlight-service': 'e2e-nextjs-winston',
	},
	level: 'debug',
})

export const logger = winston.createLogger({
	level: 'info',
	format: winston.format.combine(
		winston.format.json(),
		winston.format.errors({ stack: true }),
		winston.format.timestamp(),
		winston.format.prettyPrint(),
	),
	transports: [new winston.transports.Console(), highlightTransport],
})

...

function handler(req: NextApiRequest, res: NextApiResponse<Data>) {
	const user = {
		name: 'vadim',
	} as any
	for (let i = 0; i < 100; i++) {
		user[i.toString()] = Math.random()
	}

	logger.info('User authed', user)
	logger.warn(`User auther stringify: ${JSON.stringify(user)}`)
}
```

as you can see the first `user` object from the info level is sent over with all attributes. if you can create a repro repository with your exact case, that would be very helpful for us to look into!

**Comment by lveillard** [comment]
Oh, my bad, I see now the issue is just that the format is weird, I expect the object as an object but it arrives as an array of properties mixed with the meta-properties of the message:

![image](https://github.com/highlight/highlight/assets/29461076/f3fe7b18-d0fb-4077-8512-bb2f505ae045)

So for instance for this
`logger.warn('hey', { a: { c: 7 }, b: 2 });`
I would expect to receive two lines:
- hey
- { a: { c: 7 }, b: 2 });
(and then the metadata)

But I get three:
![image](https://github.com/highlight/highlight/assets/29461076/49910d41-9955-4b91-b428-8df1de0e8ac2).

Another example weird situation would be to print this:
`logger.warn({ level: 'success', message: 'my message' });`
Look at the result 🤪
![image](https://github.com/highlight/highlight/assets/29461076/4a64092c-f687-4156-8ce5-e283a3ddc3eb)

Btw in your image i can see some duplicated properties not sure if its normal
![image](https://github.com/highlight/highlight/assets/29461076/175e73e8-0f0f-4c25-ac73-bec9055fd9f2)

## Expected behaviour:
- So basically I would split the data in the logger in two blocks: content & metadata and not mix them.
- I would present object as objects from their root element. Right now they become objects in the frontend only from the first nested level of the obj

**Comment by lveillard** [comment]
Btw not hardcoding the levels would be great! I do actually have a level:"success" and it gets just printed as "info" 😢
